client/extensions/woocommerce/components/api-keys-view/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormInputValidation from 'components/forms/form-input-validation';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+
+class APIKeysView extends Component {
+	static propTypes = {
+		highlightEmptyFields: PropTypes.bool,
+		onEdit: PropTypes.func.isRequired,
+		keys: PropTypes.arrayOf(
+			PropTypes.shape( {
+				id: PropTypes.string.isRequired,
+				label: PropTypes.string,
+				placeholder: PropTypes.string,
+				value: PropTypes.string
+			} )
+		),
+	}
+
+	renderOneKey = ( key ) => {
+		const { highlightEmptyFields, onEdit, translate } = this.props;
+		const isError = highlightEmptyFields && ! key.value.trim();
```

Comment:
Random side note, and not sure if this validation is done server side, but we could potentially consider testing keys as they are entered client-side... some ideas around that [here](https://stackoverflow.com/questions/16388204/stripe-validating-publishable-and-secret-api-keys). But definitely follow-up PR territory there ;)